### ISSUE TITLE
es.default.config remains in start script when upgrading from 1.x to 2.x on Ubuntu 14.04

### ISSUE THREAD

**Issue by mzhaase** [elastic/puppet-elasticsearch#637]
We had 1.7.5 installed on Ubuntu Server 14.04 with the following puppet conf:

```
package { 'openjdk-7-jre': ensure => 'installed' } ->
  class { 'elasticsearch':
    ensure      => 'present',
    package_url => "https://download.elasticsearch.org/elasticsearch/elasticsearch/elasticsearch-1.7.5.deb",
    config      => {
      'discovery.zen.ping.multicast.enable'  => false,
      'discovery.zen.ping.unicast.hosts'     => "${::fqdn}:9300",
    },
  }->
  elasticsearch::instance { 'grafana':
    config => {
      'path.repo'              => '/var/backup/elasticsearch',
      'node.name'              => 'grafana-dash',
      'cluster.name'           => 'grafana',
      'network.host'           => '0.0.0.0',
      'transport.tcp.port'     => '9301',
      'http.port'              => '9201',
      'http.cors.enabled'      => true,
      'http.cors.allow-origin' => "'/https?:\\/\\/(grafana|localhost)(\\.domain\\.local)?(:[0-9]+)?/'",
    }
  }
```

There are more instances, which look pretty much identical so I omitted them.

Now we upgraded by changing the config like this:

```
class { 'elasticsearch':
    ensure      => 'present',
    version     => '2.3.3',
    manage_repo => true,
    repo_version => '2.x',
    config      => {
      'discovery.zen.ping.multicast.enable'  => false,
      'discovery.zen.ping.unicast.hosts'     => "${::fqdn}:9300",
    },
  }
```

This does not work however, the logs show:

`[2016-05-20 12:03:17,652][INFO ][bootstrap                ] es.default.config is no longer supported. elasticsearch.yml must be placed in the config directory and cannot be renamed.`

Because in the start script /etc/init.d/elasticsearch-grafana:

`DAEMON_OPTS="-d -p $PID_FILE -Des.default.config=$CONF_FILE -Des.default.path.home=$ES_HOME -Des.default.path.logs=$LOG_DIR -Des.default.path.data=$DATA_DIR -Des.default.path.work=$WORK_DIR -Des.default.path.conf=$CONF_DIR"`

removing the 

`-Des.default.config=$CONF_FILE`

makes it work again.

Thanks!

**Comment by tylerjl** [comment]
Thanks for the report, can I get what version of Puppet and version of this module you're running?

**Comment by mzhaase** [comment]
Hello,

puppet: 4.3.2
elasticsearch-elasticsearch: 0.10.3

**Comment by tylerjl** [comment]
It looks like this was solved back in commit 194a2af:

```
diff --git a/templates/etc/init.d/elasticsearch.Debian.erb b/templates/etc/init.d/elasticsearch.Debian.erb
index 3b78eeb..7138802 100644
--- a/templates/etc/init.d/elasticsearch.Debian.erb
+++ b/templates/etc/init.d/elasticsearch.Debian.erb
@@ -104,7 +104,7 @@ fi
 # Define other required variables
 PID_FILE=/var/run/$NAME.pid
 DAEMON=$ES_HOME/bin/elasticsearch
-DAEMON_OPTS="-d -p $PID_FILE -Des.default.config=$CONF_FILE -Des.default.path.home=$ES_HOME -Des.default.path.logs=$LOG_DIR -Des.default.path.data=$DATA_DIR -Des.default.path.work=$WORK_DIR -Des.default.path.conf=$CONF_DIR"
+DAEMON_OPTS="-d -p $PID_FILE -Des.default.path.home=$ES_HOME -Des.default.path.logs=$LOG_DIR -Des.default.path.data=$DATA_DIR -Des.default.path.work=$WORK_DIR -Des.default.path.conf=$CONF_DIR"
```
According to git this should be solved in all versions since `0.10.0` - the string `default.config` doesn't exist in the module as of `0.10.3` so I can't figure out how it got in your init script. Did the module version change at some point, or maybe caching somewhere on the puppetmaster?

**Comment by mzhaase** [comment]
Hello,

yes it was indeed an old cached version. Apologies.
